CMakeLists.txt

```diff
@@ -299,7 +299,10 @@ if(CMAKE_C_COMPILER_ID MATCHES "Clang")
 elseif(CMAKE_C_COMPILER_ID MATCHES "GNU")
     message(STATUS "libavif: Enabling warnings for GCC")
     add_compile_options(-Wall -Wextra)
-elseif(CMAKE_C_COMPILER_ID MATCHES "MSVC")
+elseif(MINGW)
```

Comment:
We should test this. I remember MINGW is GCC, so I am worried that it will match the GCC check at line 299 instead.

Reply:
Definitely, let me add a MINGW workflow from https://github.com/msys2/setup-msys2

Reply:
If MINGW is identified as GCC, we will need to handle it as follows:

```
elseif(CMAKE_C_COMPILER_ID MATCHES "GNU")
    message(STATUS "libavif: Enabling warnings for GCC")
    add_compile_options(-Wall -Wextra)
    if(MINGW)
       add_compile_options(-municode)
    endif()
elseif(MSVC)
```

Reply:
On the Web I found at least two pages that say `-municode` can be used as a linker option. Here is a patch file to do that:

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 12e0a10..6b05fb7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -802,11 +802,17 @@ if(AVIF_BUILD_APPS)
     if(AVIF_USE_CXX)
         set_target_properties(avifenc PROPERTIES LINKER_LANGUAGE "CXX")
     endif()
+    if(MINGW)
+        target_link_options(avifenc PRIVATE -municode)
+    endif()
     target_link_libraries(avifenc avif_apps)
     add_executable(avifdec apps/avifdec.c)
     if(AVIF_USE_CXX)
         set_target_properties(avifdec PROPERTIES LINKER_LANGUAGE "CXX")
     endif()
+    if(MINGW)
+        target_link_options(avifdec PRIVATE -municode)
+    endif()
     target_link_libraries(avifdec avif_apps)

     if(NOT SKIP_INSTALL_APPS AND NOT SKIP_INSTALL_ALL)
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index d174e05..b8d4b10 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -223,6 +223,9 @@ if(AVIF_BUILD_APPS)
     # When building apps, test the avifenc/avifdec.
     # 'are_images_equal' is used to make sure inputs/outputs are unchanged.
     add_executable(are_images_equal gtest/are_images_equal.cc)
+    if(MINGW)
+        target_link_options(are_images_equal PRIVATE -municode)
+    endif()
     target_link_libraries(are_images_equal aviftest_helpers)
     add_test(NAME test_cmd COMMAND bash ${CMAKE_CURRENT_SOURCE_DIR}/test_cmd.sh ${CMAKE_BINARY_DIR}
                                    ${CMAKE_CURRENT_SOURCE_DIR}/data

```

Reply:
Some webpages also say if `wmain()` is defined in a C++ file, it needs to be declared as `extern "C"` for the g++ in MinGW. In my testing I don't need to make that change to are_images_equal.cc.

Reply:
FYI, MINGW is defined for both GCC and Clang. See

https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Platform/Windows-GNU.cmake
https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Platform/Windows-Clang.cmake

In the CI, you can test both [UCRT64 and CLANG64 environments](https://www.msys2.org/docs/environments/).